### PR TITLE
Disable dependabot PRs for all but CLI plugins

### PR DESCRIPTION
**Pull request by stmcginnis**
## What this PR does / why we need it

This disables the creation of dependency update PRs from dependabot for
all but our local CLI plugins and our GitHub Actions until we have had a
chance to review everything and make sure we are only alerting on the
things that matter and that we have control over.

### CODE REVIEW

File: .github/dependabot.yml [hunk 1]
```diff
@@ -8,83 +8,107 @@ updates:
       interval: "daily"
 
   - package-ecosystem: "gomod"
+    # TODO: Disable for now, need to review ignore list and reenable later.
+    open-pull-requests-limit: 0
```
Comment (joshrosso) [inline .github/dependabot.yml:12]: i'm ok with this one being on.

File: .github/dependabot.yml [hunk 2]
```diff
@@ -8,83 +8,107 @@ updates:
...
     directory: "hack/builder"
     schedule:
       interval: "daily"
     ignore:
       - dependency-name: "tanzu-framework"
 
   - package-ecosystem: "gomod"
+    # TODO: Disable for now, need to review ignore list and reenable later.
+    open-pull-requests-limit: 0
```
Comment (joshrosso) [inline .github/dependabot.yml:21]: i'm ok with this one being on.

File: .github/dependabot.yml [hunk 3]
```diff
@@ -8,83 +8,107 @@ updates:
...
     directory: "test/"
     schedule:
       interval: "daily"
     ignore:
       - dependency-name: "tanzu-framework"
 
   - package-ecosystem: "gomod"
+    # TODO: Disable for now, need to review ignore list and reenable later.
+    open-pull-requests-limit: 0
```
Comment (joshrosso) [inline .github/dependabot.yml:30]: i'm ok with this one being on.

File: .github/dependabot.yml [hunk 4]
```diff
@@ -8,83 +8,107 @@ updates:
...
     directory: "hack/asset"
     schedule:
       interval: "daily"
     ignore:
       - dependency-name: "tanzu-framework"
 
   - package-ecosystem: "gomod"
+    # TODO: Disable for now, need to review ignore list and reenable later.
+    open-pull-requests-limit: 0
```
Comment (joshrosso) [inline .github/dependabot.yml:39]: i'm ok with this one being on.

File: .github/dependabot.yml [hunk 5]
```diff
@@ -8,83 +8,107 @@ updates:
...
     directory: "hack/tools"
     schedule:
       interval: "daily"
     ignore:
       - dependency-name: "tanzu-framework"
 
   - package-ecosystem: "gomod"
+    # TODO: Disable for now, need to review ignore list and reenable later.
+    open-pull-requests-limit: 0
```
Comment (joshrosso) [inline .github/dependabot.yml:48]: i'm ok with this one being on.

File: .github/dependabot.yml [hunk 6]
```diff
@@ -8,83 +8,107 @@ updates:
...
     directory: "hack/tagger"
     schedule:
       interval: "daily"
     ignore:
       - dependency-name: "tanzu-framework"
 
   - package-ecosystem: "gomod"
+    # TODO: Disable for now, need to review ignore list and reenable later.
+    open-pull-requests-limit: 0
     directory: "hack/packages"
```
Comment (joshrosso) [inline .github/dependabot.yml:58]: i'm ok with this one being on.

File: .github/dependabot.yml [hunk 7]
```diff
@@ -8,83 +8,107 @@ updates:
...
     schedule:
       interval: "daily"
     ignore:
       - dependency-name: "tanzu-framework"
 
   - package-ecosystem: "gomod"
+    # TODO: Disable for now, need to review ignore list and reenable later.
+    open-pull-requests-limit: 0
```
Comment (joshrosso) [inline .github/dependabot.yml:66]: i'm ok with this one being on.

File: .github/dependabot.yml [hunk 8]
```diff
@@ -8,83 +8,107 @@ updates:
...
     directory: "hack/dailybuild"
     schedule:
       interval: "daily"
     ignore:
       - dependency-name: "tanzu-framework"
 
   - package-ecosystem: "gomod"
+    # TODO: Disable for now, need to review ignore list and reenable later.
+    open-pull-requests-limit: 0
```
Comment (joshrosso) [inline .github/dependabot.yml:75]: i'm ok with this one being on.

File: .github/dependabot.yml [hunk 9]
```diff
@@ -8,83 +8,107 @@ updates:
...
     directory: "hack/imagelinter"
     schedule:
       interval: "daily"
     ignore:
       - dependency-name: "tanzu-framework"
 
   - package-ecosystem: "gomod"
+    # TODO: Disable for now, need to review ignore list and reenable later.
+    open-pull-requests-limit: 0
```
Comment (joshrosso) [inline .github/dependabot.yml:84]: i'm ok with this one being on.

File: .github/dependabot.yml [hunk 10]
```diff
@@ -8,83 +8,107 @@ updates:
...
     directory: "hack/runner/webhook"
     schedule:
       interval: "daily"
     ignore:
       - dependency-name: "tanzu-framework"
 
   - package-ecosystem: "gomod"
+    # TODO: Disable for now, need to review ignore list and reenable later.
+    open-pull-requests-limit: 0
```
Comment (joshrosso) [inline .github/dependabot.yml:93]: i'm ok with this one being on.

File: .github/dependabot.yml [hunk 11]
```diff
@@ -8,83 +8,107 @@ updates:
...
     directory: "hack/release/release"
     schedule:
       interval: "daily"
     ignore:
       - dependency-name: "tanzu-framework"
 
   - package-ecosystem: "gomod"
+    # TODO: Disable for now, need to review ignore list and reenable later.
+    open-pull-requests-limit: 0
```
Comment (joshrosso) [inline .github/dependabot.yml:102]: i'm ok with this one being on.

File: .github/dependabot.yml [hunk 12]
```diff
@@ -8,83 +8,107 @@ updates:
...
     directory: "hack/workflows/packages"
     schedule:
       interval: "daily"
     ignore:
       - dependency-name: "tanzu-framework"
 
   - package-ecosystem: "gomod"
+    # TODO: Disable for now, need to review ignore list and reenable later.
+    open-pull-requests-limit: 0
```
Comment (joshrosso) [inline .github/dependabot.yml:111]: i'm ok with this one being on.

**Review by joshrosso — CHANGES_REQUESTED** [review]
I'm ok with ~50% of these.

I primarily don't want to auto-open PRs for:

* `addons/packages/*`

Reason being:

* I don't want a change in a common testing dep to open a PR for every package version of a package
* Our team doesn't own these deps

hope this helps/makes sense!

File: .github/dependabot.yml (hunk 12 above)
Comment (jpmcb) [reply .github/dependabot.yml:111]: FYI - This is the shared library for testing between the different packages; I agree, I think it makes sense to keep dependabot on for it but seems to be a "gray area" for ownership between our bits and the `addons/packages` bits package owners own.

**Review by jpmcb — COMMENTED** [review]
I agree with Josh; we can own the `hack/` and other dependencies but should turn off the ones for `addons/packages`.

**Review by joshrosso — APPROVED** [review]
looks great, thanks @stmcginnis !

**Pull request merged**